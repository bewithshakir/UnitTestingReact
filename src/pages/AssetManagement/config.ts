import { IDynamicFilterProps } from "../../components/UIComponents/RightInfoPanel/DynamicFilterContent.component";

export const sortByOptions = [
    "assetManagement.sortBy.payment in progress",
    "assetManagement.sortBy.recently added lots"
  ];


export const filterByFields: IDynamicFilterProps['fields'] = [
    { name: 'state', label: 'customer-filter-panel.state', fieldType: 'select', optionUrlKey: 'assetFilter', optionAPIResponseKey: 'states', initialValue: [] },
    { name: 'city', label: 'customer-filter-panel.city', fieldType: 'select', optionUrlKey: 'assetFilter', optionAPIResponseKey: 'cities', initialValue: [] }
  ];