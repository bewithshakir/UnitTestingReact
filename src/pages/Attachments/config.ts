export const config = {
    maxAllowedFileSizeBtyes : 25000000,
    acceptedFiles: '.pdf,.docx,.xlsx',
    noOfFilesLimit: 1,
    multipleAllowed: false
};