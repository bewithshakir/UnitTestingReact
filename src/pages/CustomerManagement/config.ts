import { IDynamicFilterProps } from "../../components/UIComponents/RightInfoPanel/DynamicFilterContent.component";

export const sortByOptions = [
  "customerManagement.sortBy.atoz",
  "customerManagement.sortBy.ztoa",
  "customerManagement.sortBy.newestToOldest",
  "customerManagement.sortBy.oldestToNew",
];

export const filterByFields: IDynamicFilterProps['fields'] = [
  { name: 'date', label: 'Date', fieldType: 'dateRange', initialValue: [null, null] },
  { name: 'state', label: 'customer-filter-panel.state', fieldType: 'select', optionUrlKey: 'customerFilter', optionAPIResponseKey: 'states', initialValue: [] },
  { name: 'city', label: 'customer-filter-panel.city', fieldType: 'select', optionUrlKey: 'customerFilter', optionAPIResponseKey: 'cities', initialValue: [] },
  { name: 'paymentType', label: 'customer-filter-panel.settlement type', fieldType: 'select', optionUrlKey: 'customerFilter', optionAPIResponseKey: 'settlementType', initialValue: [] },
];