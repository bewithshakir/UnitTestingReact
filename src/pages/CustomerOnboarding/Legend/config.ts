export const config = [
    {
        index: 1,
        label: "legend.customerProfile",
        secondaryText: false,
        to: "/customer/addCustomer"
    },
    {
        index: 2,
        label: "legend.parkingLots",
        secondaryText: true,
        to: "/customer/parkingLots"
    },
    {
        index: 3,
        label: "legend.vehicles",
        secondaryText: true,
        to: "/customer/vehicles"
    },
    {
        index: 4,
        label: "legend.dsp",
        secondaryText: true,
        to: "/customer/dsps"
    },
    {
        index: 5,
        label: "legend.users",
        secondaryText: true,
        to: "/users"
    },
    {
        index: 6,
        label: "legend.attachments",
        secondaryText: true,
        to: '/customer/attachments'
    }
];

export const boxSystem = {
    border: 1
};

