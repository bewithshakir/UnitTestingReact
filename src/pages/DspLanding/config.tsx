

export const filterByFields = [];



export const SORTBY_TYPES = {
    DSP_NAME_AZ: "DSP Name A-Z",
    DSP_NAME_ZA: "DSP Name Z-A",
  };

export const DSPLanding = {
    LandingPage: {
      SortByOptions: [
        "dsp.sortBy.dspname_atoz",
        "dsp.sortBy.dspname_ztoa",
      ]
    }
  };
