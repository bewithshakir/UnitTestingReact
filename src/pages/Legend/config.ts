export const config = [
    {
        index:1,
        label: "legend.customerProfile"
    },
    {
        index:2,
        label: "legend.parkingLots"
    },
    {
        index:3,
        label: "legend.vehicles"
    },
    {
        index:4,
        label: "legend.users"
    },
    {
        index:5,
        label: "legend.dsp"
    },
    {
        index:6,
        label: "legend.attachments"
    }
]

export const boxSystem = {
    width: 276, 
    height: 375, 
    border: 1
}
