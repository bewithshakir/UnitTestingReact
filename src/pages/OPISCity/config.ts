import { ExportIcon } from '../../assets/icons';
import { IDynamicFilterProps } from '../../components/UIComponents/RightInfoPanel/DynamicFilterContent.component';

export interface IAddOPISCity{
  countryCd: string,
  state: string,
  city: string,
  cityId: string
}

export const formStatusObj = {
  success: {
    message: 'Product is successfully added',
    type: 'Success',
  },
  error: {
    message: 'Something went wrong. Please try again.',
    type: 'Error',
  },
};

export const MASS_ACTION_TYPES = {
  EXPORT: 'export',
};

export const SORTBY_TYPES = {
  CITY_NAME_AZ: "City Name A-Z",
  CITY_NAME_ZA: "City Name Z-A",
};

export const ROW_ACTION_TYPES = {
  EDIT: 'edit'
};

export const FilterByFields: IDynamicFilterProps['fields'] = [
  { name: 'state', label: 'taxes.opisCities.filter.state', fieldType: 'select', optionUrlKey: 'opisCityFilter', optionAPIResponseKey: 'states', initialValue: [] },
  { name: 'city', label: 'taxes.opisCities.filter.city', fieldType: 'select', optionUrlKey: 'opisCityFilter', optionAPIResponseKey: 'cities', initialValue: [] },
];

export const OPISCity = {
  LandingPage: {
    SortByOptions: [
      "taxes.opisCities.sortBy.cityname_atoz",
      "taxes.opisCities.sortBy.cityname_ztoa",
    ],
    FilterByFields,
    MassActionOptions: [
      {
        label: "menus.actions.export data",
        icon: ExportIcon,
        action: MASS_ACTION_TYPES.EXPORT
      },
    ],
    RowActionsOptions: [
      {
        label: "menus.data-grid-actions.edit",
        action: ROW_ACTION_TYPES.EDIT
      },
    ],
    DataGridFields: {
      "CITYID": {
        field: "cityId", label: "CITY ID"
      },
      "CITY": {
        field: "cityName", label: "CITY"
      },
      "STATE": {
        field: "stateName", label: "STATE"
      },
    },
  }
};
