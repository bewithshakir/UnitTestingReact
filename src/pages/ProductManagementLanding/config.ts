import { IDynamicFilterProps } from "../../components/UIComponents/RightInfoPanel/DynamicFilterContent.component";

export const sortByOptions = [
    //TO DO
  ];

export const filterByFields: IDynamicFilterProps['fields'] = [
  { name: 'state', label: 'customer-filter-panel.state', fieldType: 'select', optionUrlKey: 'salesTaxFilter', optionAPIResponseKey: 'states', initialValue: [] },
  { name: 'city', label: 'customer-filter-panel.city', fieldType: 'select', optionUrlKey: 'salesTaxFilter', optionAPIResponseKey: 'cities', initialValue: [] }
];
