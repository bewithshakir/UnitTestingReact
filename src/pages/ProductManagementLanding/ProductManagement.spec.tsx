import * as React from 'react';
import { shallow } from 'enzyme';
import ProductModel from '../../models/ProductManagementModel';
import ActionsMenu from '../../components/UIComponents/Menu/ActionsMenu.component';
import SortbyMenu from '../../components/UIComponents/Menu/SortbyMenu.component';
import SearchInput from "../../components/UIComponents/SearchInput/SearchInput";
import { ProductManagement } from "./config";

jest.mock("react-query", () => {
    return {
        useInfiniteQuery: jest.fn(),
    };
});

jest.mock("./queries", () => {
    return {
        ProductsListSet: jest.fn(),
    };
});


describe('Given Bruger Menu on Product Landing Page', () => {
    const CustomerObj = new ProductModel();
    const massActionOptions = CustomerObj.massActions();
    test('Render Bruger Menu', () => {
        const BurgerMenu = shallow(
            <ActionsMenu
                options={massActionOptions}
                onSelect={() => jest.fn()}
            />
        );
        expect(BurgerMenu).toMatchSnapshot();
    });

    test('Bruger Menu Menu With Options', () => {
        const BurgerMenu = shallow(
            <ActionsMenu
                options={massActionOptions}
                onSelect={() => jest.fn()}
            />
        );
        BurgerMenu.find(".btn-listmemu").simulate('click');
        expect(BurgerMenu.find(".btn-listmemu").hasClass('active')).toBe(true);
        expect(BurgerMenu.find('.actions-popper').exists()).toBe(true);
    });

});

describe('Given Sortby Menu on Product Landing Page', () => {
    test('Render Sortby Menu', () => {
        const CSortbyMenu = shallow(
            <SortbyMenu
                options={ProductManagement.LandingPage.SortByOptions.map((sortByItem) => (sortByItem))}
                onSelect={() => jest.fn()}
            />
        );
        expect(CSortbyMenu).toMatchSnapshot();
    });

    test('Sortby Menu Menu With Options', () => {
        const CSortbyMenu = shallow(
            <SortbyMenu
                options={ProductManagement.LandingPage.SortByOptions.map((sortByItem) => (sortByItem))}
                onSelect={() => jest.fn()}
            />
        );
        CSortbyMenu.find(".btn-sortby").simulate('click');
        expect(CSortbyMenu.find(".btn-sortby").hasClass('active')).toBe(true);
        expect(CSortbyMenu.find('.sortby-popper').exists()).toBe(true);
    });

});

describe('Given Search Input on Product Landing Page', () => {
    test('Render Search Input onChange', () => {
        const searchInputItem = shallow(
            <SearchInput
                onChange={() => jest.fn()}
            />
        );
        expect(searchInputItem).toMatchSnapshot();
    });

    test('Search Input onBlur', () => {
        const searchInputItem = shallow(
            <SearchInput
                onChange={() => jest.fn()}
                onBlur={() => jest.fn()}
            />
        );
        searchInputItem.find(".searchinput").simulate('click');
        expect(searchInputItem.find('.adornment')).toBeDefined();
    });

});