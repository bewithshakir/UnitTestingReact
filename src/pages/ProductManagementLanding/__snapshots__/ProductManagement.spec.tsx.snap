// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Bruger Menu on Product Landing Page Render Bruger Menu 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    aria-haspopup="true"
    aria-label="actions menu list"
    className="btn-listmemu "
    onClick={[Function]}
    size="small"
    startIcon={
      <WithStyles(ForwardRef(SvgIcon))
        component={[Function]}
        viewBox="0 0 20 20"
      />
    }
    variant="contained"
  />
  <ForwardRef(Popper)
    anchorEl={null}
    className="actions-popper"
    disablePortal={true}
    open={false}
    placement="bottom-end"
    transition={true}
  >
    <Component />
  </ForwardRef(Popper)>
</div>
`;

exports[`Given Search Input on Product Landing Page Render Search Input onChange 1`] = `
<ForwardRef(OutlinedInput)
  className="searchinput"
  id="SearchInput"
  inputProps={
    Object {
      "aria-label": "search",
    }
  }
  onChange={[Function]}
  placeholder="Search"
  startAdornment={
    <ForwardRef(InputAdornment)
      className="adornment"
      position="start"
    >
      <ForwardRef(IconButton)
        aria-label="search"
        edge="start"
      >
        <DropSearchIcon />
      </ForwardRef(IconButton)>
    </ForwardRef(InputAdornment)>
  }
  style={
    Object {
      "width": "100%",
    }
  }
  type="text"
  value=""
/>
`;

exports[`Given Sortby Menu on Product Landing Page Render Sortby Menu 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    aria-haspopup="true"
    aria-label="sortby menu list"
    className="btn-sortby "
    onClick={[Function]}
    startIcon={<Memo />}
  >
    buttons.sort by
  </WithStyles(ForwardRef(Button))>
  <ForwardRef(Popper)
    anchorEl={null}
    className="sortby-popper"
    data-testid="sortby-dropdown"
    disablePortal={true}
    open={false}
    placement="bottom-start"
    transition={true}
  >
    <Component />
  </ForwardRef(Popper)>
</div>
`;
